apps/server/src/controllers/paymentsController/fundGroupSavingsWallet.ts

```diff
@@ -0,0 +1,122 @@
+    const reqData = fundGroupWalletSchema.strict().safeParse(verifyAmount)
+
+    if (!reqData.success) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: reqData.error.issues,
+      })
+    }
+
+    const { amount } = reqData.data
+
+    const groupSavingsWallet = await Wallets.findOne({
+      where: {
+        [Op.and]: [
+          { ownerId: userId },
+          { ownerType: OwnerType.USER },
+          { type: WalletType.GROUP_WALLET },
+        ],
+      },
+    })
+    const globalWallet = await Wallets.findOne({
+      where: {
+        [Op.and]: [
+          { ownerId: userId },
+          { ownerType: OwnerType.USER },
+          { type: WalletType.GLOBAL },
+        ],
+      },
+    })
+
+    if (globalWallet && groupSavingsWallet) {
+      if (globalWallet.totalAmount < amount) {
+        return res.status(HTTP_STATUS_CODE.FORBIDDEN).json({
+          message: 'Insufficient funds in your global wallet',
+        })
+      }
+      const newGlobalBalance = globalWallet.totalAmount - amount
+      const newGroupSavingsBalance = groupSavingsWallet.totalAmount + amount
+      const newGoupSavingsIncome = groupSavingsWallet.totalIncome + amount
+      const newIncome = {
+        amount,
+        date: new Date(),
+      }
+      const newEarnings = [...groupSavingsWallet.earnings, newIncome]
+
+      globalWallet.totalAmount = newGlobalBalance
+      const newGlobalWallet = await globalWallet.save()
+
+      groupSavingsWallet.totalAmount = newGroupSavingsBalance
+      groupSavingsWallet.totalIncome = newGoupSavingsIncome
+      groupSavingsWallet.earnings = newEarnings
+      const newGroupSavingsWallet = await groupSavingsWallet.save()
+
+      if (newGroupSavingsWallet && newGlobalWallet) {
+        //create successful debit  transaction
+        const globalGroupTransactionDetails = {
+          walletId: newGlobalWallet.id,
+          ownerId: userId,
+          amount: amount,
+          status: transactionStatus.SUCCESSFUL,
+          action: action.DEBIT,
+          type: transactionType.INTERNAL_TRANSFER,
+          receiverId: newGroupSavingsWallet.id,
+          createdAt: newGroupSavingsWallet.createdAt,
+          updatedAt: newGroupSavingsWallet.updatedAt,
+        }
+        const groupGlobalDebit = await createTransaction(
+          globalGroupTransactionDetails
+        )
+        //create successfull credit transaction
+        const savingsGroupTransDetails = {
+          walletId: newGroupSavingsWallet.id,
+          ownerId: userId,
+          amount: amount,
+          status: transactionStatus.SUCCESSFUL,
+          action: action.CREDIT,
+          type: transactionType.INTERNAL_TRANSFER,
+          senderId: newGlobalWallet.id,
+          createdAt: newGlobalWallet.createdAt,
+          updatedAt: newGlobalWallet.updatedAt,
+        }
+        const groupSavingsTrans = await createTransaction(
+          savingsGroupTransDetails
+        )
```

Comment:
The if statement shows that there's a possibility for global wallet or groupSavings wallet not to be found. You didn't handle that because there's no else block.

Reply:
The else block is optional. If you don't include it, the program will simply move on to the next statement after the if block